crypto/curve25519/curve25519.c

```diff
@@ -301,37 +301,31 @@ int ED25519_sign(uint8_t out_sig[64], const uint8_t *message,
 
 int ED25519_verify(const uint8_t *message, size_t message_len,
                    const uint8_t signature[64], const uint8_t public_key[32]) {
-  ge_p3 A;
-  if ((signature[63] & 224) != 0 ||
-      !x25519_ge_frombytes_vartime(&A, public_key)) {
-    return 0;
-  }
 
-  fe_loose t;
-  fe_neg(&t, &A.X);
-  fe_carry(&A.X, &t);
-  fe_neg(&t, &A.T);
-  fe_carry(&A.T, &t);
-
-  uint8_t pkcopy[32];
-  OPENSSL_memcpy(pkcopy, public_key, 32);
-  uint8_t rcopy[32];
-  OPENSSL_memcpy(rcopy, signature, 32);
-  uint8_t scopy[32];
-  OPENSSL_memcpy(scopy, signature + 32, 32);
+  // Ed25519 verify: rfc8032 5.1.7
 
-  // https://tools.ietf.org/html/rfc8032#section-5.1.7 requires that s be in
-  // the range [0, order) in order to prevent signature malleability.
+  // Step: rfc90832 5.1.7.1
+  // Decode signature as:
+  //  - signature[0:31]: encoded point R, aliased to R_expected.
+  //  - signature[32:61]: integer S.
+  if ((signature[63] & 224) != 0) {
```

Comment:
What's the purpose of the magic value `224` out of curiosity? Didn't find it in the RFC at a quick glance.

Reply:
I understand it's coming from this guarantee when signing in 5.1.6.6:
> Form the signature of the concatenation of R (32 octets) and the little-endian encoding of S (32 octets; the three most significant bits of the final octet are always zero).

But I didn't dive into the exact details of encoding.
It's worth adding a comment.

Reply:
5.1.6.6 correct. We have 224 = 0xE0